google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerTableStatements.java

```diff
@@ -53,17 +57,38 @@ public SpannerTableStatements(SpannerDialect spannerDialect) {
    */
   public List<String> dropTable(Table table, Metadata metadata) {
     ArrayList<String> dropStrings = new ArrayList<>();
-
-    Iterator<Index> iteratorIdx = table.getIndexIterator();
-    while (iteratorIdx.hasNext()) {
-      Index curr = iteratorIdx.next();
-      dropStrings.add("drop index " + curr.getName());
+    for (String indexName : getTableIndices(table)) {
+      dropStrings.add("drop index " + indexName);
     }
 
     dropStrings.add(this.spannerDialect.getDropTableString(table.getQuotedName()));
     return dropStrings;
   }
 
+  private Set<String> getTableIndices(Table table) {
+    Set<String> tableIndices = new HashSet<>();
+
+    Iterator<Index> indexIterator = table.getIndexIterator();
+    while (indexIterator.hasNext()) {
+      tableIndices.add(indexIterator.next().getName());
+    }
+
+    Iterator<UniqueKey> keyIterator = table.getUniqueKeyIterator();
+    while (keyIterator.hasNext()) {
+      tableIndices.add(keyIterator.next().getName());
+    }
+
+    Iterator<Column> colIterator = table.getColumnIterator();
+    while (colIterator.hasNext()) {
+      Column col = colIterator.next();
+      if (col.isUnique()) {
+        tableIndices.add(Constraint.generateName("UK_", table, col));
```

Comment:
This won't show up in table.getUniqueKeyIterator() earlier? It was added in `table.getOrCreateUniqueKey(name)` during initialization.

Reply:
Nice catch, this is not needed. Removed.